File: quasar-front/src/app/shared/models/requests/produtoBuscar.request.ts
import { PaginacaoRequest } from './paginacao.request';

export class ProdutoBuscarRequest extends PaginacaoRequest {

  codigo: ( number | string ) ;
  nome: string;
  codCategoria: ( number | string );

  constructor(params: Partial<ProdutoBuscarRequest>) {
    super(params);
    this.codigo = params.codigo || '';
    this.nome = params.nome || '';
    this.codCategoria = params.codCategoria || '';
  }
}

File: quasar/Quasar.DataTransfer/Produtos/Requests/ProdutoBuscarRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Quasar.DataTransfer.Genericos.Requests;

namespace Quasar.DataTransfer.Produtos.Requests
{
    public class ProdutoBuscarRequest : PaginacaoRequest
    {
        public int? Codigo { get; set; }
        public string? Nome { get; set; }
        public int? CodCategoria { get; set; }
    }
}
